### PR TITLE
Fix machine boot into xen hyperv after installing with kvm autoyast profile

### PR DESCRIPTION
**Pull request by varunkojha**
Description 
-- Autoyast allows the partitioning and formatting of disks during the installation process. We can specify the partitioning disks and list them. This will ensure that the Autoyast profile references the correct disk for installation and avoid booting the machine into the XEN hypervisor. 

-- The changes I've made to the AutoYast profile involve adding an additional drive (/dev/sdb) to the partitioning configuration. This change allows the installer to recognize and initialize both /dev/sda and /dev/sdb during the installation process. 

-- This is implemented in KVM autoyast profile at the partitioning drive. The original file is updated in /var/lib/openqa/share/factory/other/autoyast_opensuse_kvm_sshd.xml on Ariel [gate.opensuse.org](http://gate.opensuse.org/) and this PR is to track in GIT. [HERE](https://openqa.opensuse.org/assets/other/autoyast_opensuse_kvm_two_disk.xml)

-- renamed: autoyast_opensuse_kvm_sshd.xml -> autoyast_opensuse_kvm_one_disk.xml
-- created: autoyast_opensuse_kvm_two_disk.xml
-- created: autoyast_opensuse_xen_two_disk.xml
-- renamed: autoyast_opensuse_xen_sshd.xml -> autoyast_opensuse_xen_one_disk.xml
-- menu.ipxe has been updated accordingly to track changes in GIT.
-- comments added in profile: note these changes are already there, updating now to track in GIT.

- Related ticket: https://progress.opensuse.org/issues/130979 
- Verification run: 
KVM: [1](https://openqa.opensuse.org/tests/3523492), [2](https://openqa.opensuse.org/tests/3523050), [3](https://openqa.opensuse.org/tests/3518992), [4](https://openqa.opensuse.org/tests/3516805), [5.](https://openqa.opensuse.org/tests/3508499)  
XEN: https://openqa.opensuse.org/tests/3533929

- Latest run: [KVM_rebel_5](https://openqa.opensuse.org/tests/3538265) and [KVM_rebel_6](https://openqa.opensuse.org/tests/3537751)

- Verification Run after file_change: [KVM_rebel5](https://openqa.opensuse.org/tests/3544979)

Welcome review!! Thanks :)
@Julie-CAO @guoxuguang @waynechen55 @alice-suse @RoyCai7 @tonyyuan1 @tbaev @nanzhg

### CODE REVIEW

**Comment by Julie-CAO** [comment]
> Tracking earlier change -- virtualization-related service virtnwfilterd is removed, and description added: note these changes are already there, updating now to track in GIT.

I can't remember that it is me that said why and when virtnwfilterd was removed? Could you give me a hint?

File: data/virt_autotest/host_unattended_installation_files/autoyast_opensuse_kvm_sshd.xml [hunk 1]
```diff
@@ -53,13 +53,19 @@
...
       <initialize t="boolean">true</initialize>
       <use>all</use>
       <enable_snapshots t="boolean">false</enable_snapshots>
     </drive>
+    <drive>
+      <device>/dev/sdb</device>
+      <initialize t="boolean">true</initialize>
+      <use>all</use>
+      <enable_snapshots t="boolean">false</enable_snapshots>
+    </drive>
```
Comment (Julie-CAO) [inline data/virt_autotest/host_unattended_installation_files/autoyast_opensuse_kvm_sshd.xml:68]: I saw there has been some successful builds with this fix.  If adding the 2nd disk in autoyast profile could resolve the long exist failure, we can take it as the final fix because it saves effort.

But the limitation of this change is that these autoyast profile could be only used for machines with 2 and more hard disks. Machines with single disk will fail to install due to missing sdb, fg. http://10.67.129.27/tests/76#step/ipxe_install/2

It is not the best solution, but it can be acceptable as we need these tests go into TW job group quickly. So how about we change the name of this file to like 'autoyast_opensuse_kvm_two_disk.xml' to flag that it is limited to machines with 2 disks. We change the iPXE bootscripts in O3 to call this 'two_disk.xm', then the two tests on O3 will work as usual.

Meanwhile we create another 'autoyast_opensuse_kvm_one_disk.xml' on the same directory.  For installing TW outside O3, We can set `AUTOYAST=xxx` in test suites to specify which profiles to be used, two-disk or one-disk, based on your own worker.

File: data/virt_autotest/host_unattended_installation_files/autoyast_opensuse_kvm_sshd.xml [hunk 2]
```diff
@@ -53,13 +53,19 @@
       <xen_append>console=hvc0,115200 console=tty loglevel=5</xen_append>
     </global>
   </bootloader>
-  <partitioning t="list">
+  <partitioning config:type="list">
```
Comment (Julie-CAO) [inline data/virt_autotest/host_unattended_installation_files/autoyast_opensuse_kvm_sshd.xml:56]: this change is not needed. `t` is the abbreation of `config:type=`.

Comment (varunkojha) [reply data/virt_autotest/host_unattended_installation_files/autoyast_opensuse_kvm_sshd.xml:56]: ok

File: data/virt_autotest/host_unattended_installation_files/autoyast_opensuse_kvm_sshd.xml (hunk 1 above)
Comment (varunkojha) [reply data/virt_autotest/host_unattended_installation_files/autoyast_opensuse_kvm_sshd.xml:68]: I kind of like this way and approach and thought process because other solutions tried earlier had one or more problems in other areas.

**Comment by varunkojha** [comment]
> > Tracking earlier change -- virtualization-related service virtnwfilterd is removed, and description added: note these changes are already there, updating now to track in GIT.
> 
> I can't remember that it is me that said why and when virtnwfilterd was removed? Could you give me a hint?

```
Julie Cao
  [3:17 PM](https://suse.slack.com/archives/D04FUTXNVQC/p1690192042838359)
without this trick, the br0 would get a different MAC and IP with em1.
we will lost ssh connection to the machine after  installation.
Varun Kumar Ojha
  [3:18 PM](https://suse.slack.com/archives/D04FUTXNVQC/p1690192137371359)
ok it will be updated in git
i have done the changes locally
Julie Cao
  [3:21 PM](https://suse.slack.com/archives/D04FUTXNVQC/p1690192305378149)
about the comments
 <!-- change MACAddressPolicy to none, or the network bridge will get a different MAC address from its enslave -->
    <!-- becasue TW uses udev naming scheme v247, see bsc#1136600 -->
It is possible I removed them from my PR by accident.
thank you for adding them back
```

**Comment by Julie-CAO** [comment]
> > > Tracking earlier change -- virtualization-related service virtnwfilterd is removed, and description added: note these changes are already there, updating now to track in GIT.
> > 
> > 
> > I can't remember that it is me that said why and when virtnwfilterd was removed? Could you give me a hint?
> 
> ```
> Julie Cao
>   [3:17 PM](https://suse.slack.com/archives/D04FUTXNVQC/p1690192042838359)
> without this trick, the br0 would get a different MAC and IP with em1.
> we will lost ssh connection to the machine after  installation.
> Varun Kumar Ojha
>   [3:18 PM](https://suse.slack.com/archives/D04FUTXNVQC/p1690192137371359)
> ok it will be updated in git
> i have done the changes locally
> Julie Cao
>   [3:21 PM](https://suse.slack.com/archives/D04FUTXNVQC/p1690192305378149)
> about the comments
>  
>     
> It is possible I removed them from my PR by accident.
> thank you for adding them back
> ```

It was about the two lines of comments, nothing to do with `virtnwfilterd`. virtnwfilterd is not used in TW, but Alice said it is used in ALP, so let's keep it in TW as well for consistence.

**Comment by varunkojha** [comment]
Noted, sure thanks

File: data/virt_autotest/host_unattended_installation_files/autoyast_opensuse_kvm_sshd.xml (hunk 1 above)
Comment (varunkojha) [reply data/virt_autotest/host_unattended_installation_files/autoyast_opensuse_kvm_sshd.xml:68]: modified the file names 
new file:   autoyast_opensuse_kvm_one_disk_sshd.xml
renamed:    autoyast_opensuse_kvm_sshd.xml -> autoyast_opensuse_kvm_two_disk_sshd.xml

Comment (varunkojha) [reply data/virt_autotest/host_unattended_installation_files/autoyast_opensuse_kvm_sshd.xml]: Julie, could you please modify it as I do not have edit permissions
Test Suites -> virt-guest-installation-kvm -> Settings -> AUTOYAST=xxx from "autoyast_opensuse_kvm_sshd.xml" to autoyast_opensuse_kvm_two_disk_sshd.xml

twodisk profile picked
2disk_selected

Comment (varunkojha) [reply data/virt_autotest/host_unattended_installation_files/autoyast_opensuse_kvm_sshd.xml]: Verification Run after file renamed: [KVM_rebel5](https://openqa.opensuse.org/tests/3544979)

Comment (varunkojha) [reply data/virt_autotest/host_unattended_installation_files/autoyast_opensuse_kvm_sshd.xml]: sure, thanks!

Comment (varunkojha) [reply data/virt_autotest/host_unattended_installation_files/autoyast_opensuse_kvm_sshd.xml]: renamed:    autoyast_opensuse_kvm_one_disk_sshd.xml -> autoyast_opensuse_kvm_one_disk.xml
renamed:    autoyast_opensuse_kvm_two_disk_sshd.xml -> autoyast_opensuse_kvm_two_disk.xml

Comment (varunkojha) [reply data/virt_autotest/host_unattended_installation_files/autoyast_opensuse_kvm_sshd.xml]: renamed:    autoyast_opensuse_xen_sshd.xml -> autoyast_opensuse_xen_two_disk.xml

File: data/virt_autotest/host_unattended_installation_files/autoyast_opensuse_kvm_two_disk.xml [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+  </bootloader>
+  <partitioning t="list">
+    <drive>
+      <device>/dev/sda</device>
+      <initialize t="boolean">true</initialize>
+      <use>all</use>
+      <enable_snapshots t="boolean">false</enable_snapshots>
+    </drive>
+    <drive>
+      <device>/dev/sdb</device>
```
Comment (alice-suse) [inline data/virt_autotest/host_unattended_installation_files/autoyast_opensuse_kvm_two_disk.xml:64]: Is the main difference with one disk kvm template this section of /dev/sdb? What will be the solution if in future we get machines with eg 4 disks?

File: data/virt_autotest/host_unattended_installation_files/autoyast_opensuse_xen_two_disk.xml [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+      <serial>serial --unit=1 --speed=115200 --word=8 --parity=no</serial>
+      <timeout t="integer">30</timeout>
+      <update_nvram>true</update_nvram>
+      <xen_kernel_append>console=com2,115200 loglvl=all guest_loglvl=all sync_console</xen_kernel_append>
+      <xen_append>console=hvc0,115200 console=tty loglevel=5</xen_append>
+    </global>
+  </bootloader>
+  <partitioning t="list">
+    <drive>
+      <device>/dev/sda</device>
```
Comment (alice-suse) [inline data/virt_autotest/host_unattended_installation_files/autoyast_opensuse_xen_two_disk.xml:58]: For xen two disk autoyast, only sda is enough? What's the difference with xen one disk autoyast?

File: data/virt_autotest/host_unattended_installation_files/autoyast_opensuse_kvm_one_disk.xml [hunk 5]
```diff
@@ -128,7 +128,8 @@
   <scripts>
     <!-- permit root login, password login and pubkeys login -->
     <!-- keep ssh session alive -->
-    <!-- /etc/ssh/sshd_config is not default installed in TW -->
```
Comment (alice-suse) [inline data/virt_autotest/host_unattended_installation_files/autoyast_opensuse_kvm_one_disk.xml:131]: Why remove this line? Sshd is installed by default now?

File: data/virt_autotest/tw_ipxe/menu.ipxe [hunk 6]
```diff
@@ -1,28 +1,38 @@
...
-dhcp
+# dhcp should be already done by boot script
+#dhcp
 
 #set variables
 set menu-timeout 30000
+set server-name netboot
+iseq ${next-server} 10.150.1.11   && set server-name openqa.opensuse.org ||
+iseq ${next-server} 10.168.192.10 && set server-name qe.nue2.suse.org ||
+iseq ${next-server} 10.162.0.1    && set server-name qanet.qa.suse.de ||
```
Comment (alice-suse) [inline data/virt_autotest/tw_ipxe/menu.ipxe:14]: Would you please give more background information why three  server-name are needed? Related with NUE1 lab move?

Comment (Julie-CAO) [reply data/virt_autotest/tw_ipxe/menu.ipxe:14]: menu.ipxe in O3 was used and controled by us on O3 only, so we tracked in git under our test directory.

However days ago, it was adapted to support both PXE and iPXE boot by tools team, and apply it on both O3 and OSD. ie. it began to be managed by tools team.

We still has the access to this file and edited it on the server directly. It's better that it is still be tracked in git to avoid accedental breakage.

File: data/virt_autotest/host_unattended_installation_files/autoyast_opensuse_kvm_one_disk.xml (hunk 5 above)
Comment (Julie-CAO) [reply data/virt_autotest/host_unattended_installation_files/autoyast_opensuse_kvm_one_disk.xml:131]: this line is a piece of comment. It should have been replaced with the two lines above it in my previous PR, but I forgot it. So It was removed in this PR.

Comment (varunkojha) [reply data/virt_autotest/host_unattended_installation_files/autoyast_opensuse_kvm_one_disk.xml:131]: Yes so i added in this PR, these were already updated in the xml on ariel

File: data/virt_autotest/host_unattended_installation_files/autoyast_opensuse_xen_two_disk.xml (hunk 4 above)
Comment (varunkojha) [reply data/virt_autotest/host_unattended_installation_files/autoyast_opensuse_xen_two_disk.xml:58]: The issue was seen only with the KVM guest installation, the xen autoyast profile has <script> which makes it boot into Xen hyperv by default `GRUB_DEFAULT=.*/GRUB_DEFAULT=2/` 

The filename has been changed to meet the naming convention and to avoid confusion, so we run two_disk.xml file for both KVM and Xen on O3 machines or machines with 2 or more disks. And when we want to install on single disk we would use `AUTOYAST=xxx_one_disk.xml` for both Xen and KVM.

When we have machine with 3 or 4 disks and if the issue arises for Xen, we can work it under <partitioning> accordingly in the future, as of now not required. 

This approach allows you to accommodate both scenarios without affecting existing tests. It ensures that tests for machines with different disk configurations can proceed without issues. However, it's important to document and communicate this change to all relevant team members and testers to ensure everyone uses the correct AutoYast profile for their specific testing needs.

File: data/virt_autotest/host_unattended_installation_files/autoyast_opensuse_kvm_two_disk.xml (hunk 3 above)
Comment (varunkojha) [reply data/virt_autotest/host_unattended_installation_files/autoyast_opensuse_kvm_two_disk.xml:64]: Yes.

When we have machines with 3 or 4 disks and if the issue arises for Xen or KVM, we can work it under <partitioning> accordingly in the future, as of now not required.

The proposed solution involves creating two separate AutoYast profiles, one for machines with two or more hard disks and another for machines with a single disk. This way, we can ensure that tests for both scenarios work correctly. 

- This approach seems suitable for the specific testing and deployment scenarios. It offers a pragmatic solution to get tests running quickly and efficiently while acknowledging hardware differences.
- I believe it's important to document the purpose and usage of each profile clearly, both for the benefit of the current team and for anyone who may work with these profiles in the future.
- Regular testing and validation of these profiles on different hardware configurations will be essential to ensure that they continue to work as expected.
- As hardware configurations evolve, it may be necessary to revisit the profile strategy and consider whether a more scalable approach is needed.

In summary, while this solution may not be the most elegant from a purely technical perspective, it appears to be a practical and effective way to address the immediate need for testing on machines with different disk configurations as we need these tests to go into TW job group quickly. Careful documentation and ongoing testing will be key to its success.

File: data/virt_autotest/host_unattended_installation_files/autoyast_opensuse_xen_two_disk.xml (hunk 4 above)
Comment (varunkojha) [reply data/virt_autotest/host_unattended_installation_files/autoyast_opensuse_xen_two_disk.xml:58]: They basically do the same work as of now.

File: data/virt_autotest/tw_ipxe/menu.ipxe (hunk 6 above)
Comment (varunkojha) [reply data/virt_autotest/tw_ipxe/menu.ipxe:14]: Thankyou Julie for mentioning it

File: data/virt_autotest/host_unattended_installation_files/autoyast_opensuse_xen_two_disk.xml (hunk 4 above)
Comment (alice-suse) [reply data/virt_autotest/host_unattended_installation_files/autoyast_opensuse_xen_two_disk.xml:58]: If the issue does not happen on xen, it seems we only need to differentiate the autoyast fo kvm on one or two disk, while for xen better keep it unified as before. HDYT?

Besides, for current kvm issue solution, you are right, at the moment, it is not required for 3 or 4 disks. The point is that this solution won't work seamlessly for any machine. It's known to not work for specific conditions that may not be rare. So, it can only be short term solution. It can be accepted this time since we won't have different machines soon. Please allow me to remind that , in our team, we'd always recommend to think about long term solution as first choice. Look forward to that in future PRs!

File: data/virt_autotest/tw_ipxe/menu.ipxe (hunk 6 above)
Comment (alice-suse) [reply data/virt_autotest/tw_ipxe/menu.ipxe:14]: okay, thanks for the info

File: data/virt_autotest/host_unattended_installation_files/autoyast_opensuse_xen_two_disk.xml (hunk 4 above)
Comment (varunkojha) [reply data/virt_autotest/host_unattended_installation_files/autoyast_opensuse_xen_two_disk.xml:58]: Is it OK for me either way for Xen like this or like before, @Julie-CAO HDYT?

Comment (Julie-CAO) [reply data/virt_autotest/host_unattended_installation_files/autoyast_opensuse_xen_two_disk.xml:58]: I don't think the two-disk issue happend on kvm host only. It was not revealed on xen most likely it does exist but we did not notice it, how do you know it did not login to a previous snapshot?
So I recommended that xen has 2-disk and 1-disk profiles files both.

Comment (Julie-CAO) [reply data/virt_autotest/host_unattended_installation_files/autoyast_opensuse_xen_two_disk.xml:58]: I don't think the two-disk issue happend on kvm host only. It was not revealed on xen most likely it does exist but we did not notice it, how do you know it did not login to a previous snapshot?
So I recommended that xen has 2-disk and 1-disk profiles files both.

Comment (alice-suse) [reply data/virt_autotest/host_unattended_installation_files/autoyast_opensuse_xen_two_disk.xml:58]: okay, sounds reasonable. This does not seem to be related with kvm or xen either.

**Comment by alice-suse** [comment]
LGTM, thanks for the fix
